tests/kernels/test_pos_encoding.py

```diff
@@ -26,6 +27,7 @@
 @pytest.mark.parametrize("seed", SEEDS)
 @pytest.mark.parametrize("device", DEVICES)
 @torch.inference_mode()
+@skipIfRocm
```

Comment:
Here is the output I got when I ran `pytest test_pos_encoding` 
The steps I used to run the tests are:
Run the docker container, and then copy the `tests` to the vllm's installation location, and run `pytest` from there:

```
root@xxx:/opt/conda/envs/py_3.9/lib/python3.9/site-packages/vllm-0.2.7+rocm603-py3.9-linux-x86_64.egg/vllm/tests/kernels# pytest test_pos_encoding.py 
=================================================================================== test session starts ===================================================================================
platform linux -- Python 3.9.18, pytest-7.3.2, pluggy-1.3.0
rootdir: /opt/conda/envs/py_3.9/lib/python3.9/site-packages/vllm-0.2.7+rocm603-py3.9-linux-x86_64.egg/vllm/tests/kernels
plugins: shard-0.1.2, xdoctest-1.1.0, xdist-3.3.1, flakefinder-1.1.0, cpp-2.3.0, hypothesis-5.35.1, rerunfailures-13.0, anyio-4.2.0
collected 1152 items                                                                                                                                                                      
Running 1152 items in this shard

test_pos_encoding.py ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 13%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 29%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 44%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 60%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 75%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 91%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss                                                                                 [100%]

==================================================================================== warnings summary =====================================================================================
../../../../torch/cuda/__init__.py:611
  /opt/conda/envs/py_3.9/lib/python3.9/site-packages/torch/cuda/__init__.py:611: UserWarning: Can't initialize NVML
    warnings.warn("Can't initialize NVML")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================ 1152 skipped, 1 warning in 2.21s =============================================================================
root@xxx:/opt/conda/envs/py_3.9/lib/python3.9/site-packages/vllm-0.2.7+rocm603-py3.9-linux-x86_64.egg/vllm/tests/kernels# 
```

Reply:
ok. I will rerun the test again using ROCm6 docker container. 
I was using Dockerfile.rocm in the `hongxiayang:skipUT` branch to build a `ROCm 5.7` image environment on MI210. In that container, the tests are not skipped and all the test for `test_pos_encoding.py` failed.

Reply:
@hongxiayang I have tried building the `ROCm 6.0` to run the `test_pos_encodings.py`, it successfully skip the test.
It could be pytorch version issue. `The ROCm 5.7` docker image is using `pytorch2.0.1`, where as `ROCm 6.0` docker images contain `pytorch2.1.1`.
Regarding to this matter, should we reimplement a decorator to skip the test if ROCm is detected?

Reply:
In rocm5.7/torch2.0.1 docker image, we just need to put the environment variable PYTORCH_TEST_WITH_ROCM=1 when running pytest as below:
```
root@xxx:/opt/conda/envs/py_3.10/lib/python3.10/site-packages/vllm-0.2.4+rocm573-py3.10-linux-x86_64.egg/vllm/tests/kernels# PYTORCH_TEST_WITH_ROCM=1 pytest test_pos_encoding.py 
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.10.13, pytest-7.4.2, pluggy-1.3.0
rootdir: /opt/conda/envs/py_3.10/lib/python3.10/site-packages/vllm-0.2.4+rocm573-py3.10-linux-x86_64.egg/vllm/tests/kernels
plugins: shard-0.1.2, xdoctest-1.1.0, xdist-3.3.1, flakefinder-1.1.0, rerunfailures-12.0, hypothesis-5.35.1, anyio-4.2.0
collected 1152 items                                                                                                                                                                                       
Running 1152 items in this shard

test_pos_encoding.py sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 15%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 32%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 49%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 66%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 83%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss    [100%]

============================================================================================= warnings summary =============================================================================================
../../../../torch/cuda/__init__.py:546
  /opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/cuda/__init__.py:546: UserWarning: Can't initialize NVML
    warnings.warn("Can't initialize NVML")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================================================================================== 1152 skipped, 1 warning in 2.95s =====================================================================================
root@xxx:/opt/conda/envs/py_3.10/lib/python3.10/site-packages/vllm-0.2.4+rocm573-py3.10-linux-x86_64.egg/vllm/tests/kernels# 
```